crates/compiler/alias_analysis/src/lib.rs

```diff
@@ -1442,7 +1442,9 @@ fn expr_spec<'a>(
             union_layout,
             ..
         } => {
-            debug_assert!(indices.len() >= 2);
+            //comment written in: 2023.11.22; indices.len() can only be >2 after trmc happens in
+            //monomorphization, tail_recursion, but that happens after alias analysis
+            debug_assert!(indices.len() == 2);
```

Comment:
I don't think this is right. Alias analysis only ever happens after all passes in mono, including monomorphization and tail recursion.